plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/delete/EqualityDeleteSets.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.delete;
+
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.StructLike;
+import org.apache.iceberg.util.StructLikeSet;
+import org.apache.iceberg.util.StructProjection;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+public class EqualityDeleteSets
+{
+    private final Map<Set<Integer>, EqualityDeleteSet> deleteSetById = new HashMap<>();
```

Comment:
It seems like you could get away just using a ` Map<Set<Integer>, StructLikeSet>` and reduce some of the complexity here?

Reply:
I think the main issue is that I want to fix the problem of the RowPredicate.and stack getting deeper as the number of delete files increases. So, I used EqualityDeleteSets to collect all the StructLikeSets. However, during this process, I found that StructLikeSet needs to correspond to Schema one by one, because Schema is needed in the later process of generating Projection. Therefore, I also need to correspond it with StructLikeSet, so I created EqualityDeleteSet.

Reply:
Gotcha, I had a chance to get back to this today and I see what you're going for now. 

I think we could make this a bit easier by making an EqualityDeleteFilter that combines all the eq deletes with the same schema, rather than trying to merge all of them into one. I took a stab at it in a commit here https://github.com/alexjo2144/trino/commit/4e36dcc4c7f5c241d711a201587709eabeac0c14

Can you take a look and let me know what you think? I don't have a big enough data set to test with.

Reply:
@alexjo2144 
I think there are two main aspects of your code that are different from mine. 
1.  You merge according to List(id), while I merge according to Set(id)
2. I merged all the EqualityDeleteFilter into one, and you get multiple EqualityDeleteFilters by ids

First, I refer to the set used by iceberg’s implementation. The schema order of deletefile should not affect its use. Refer: https://github.com/apache/iceberg/blob/5c85dd345934f0148bf3c36bb563f9a89c31dfb6/data/src/main/java/org/apache/iceberg/data/DeleteFilter.java#L157-L201

Second, I think `RowPredicate.and` will generate a stack of corresponding depth according to the number of delete filters. I merged it into one is to avoid possible deep stack problems.